suitcase/json_metadata/__init__.py

```diff
@@ -8,30 +11,215 @@
+        ``json.dump`` as a kwarg which ensures that all items are encoded
+        correctly. The defualt is ``NumpyEncoder``, defined below, which also
+        ensures that all ``numpy`` objects are converted to built-in python
+        ones.
+    **kwargs : kwargs
+        kwargs to be passed to ``json.dump``.
+    Returns
+    -------
+    dest : dict
+        dict mapping the 'labels' to lists of file names
+    """
+    def __init__(self, directory, file_prefix='{uid}-', cls=NumpyEncoder,
+                 **kwargs):
+
         if isinstance(directory, (str, Path)):
             self.manager = suitcase.utils.MultiFileManager(directory)
         else:
             self.manager = directory
-        self.artifacts = self.manager._artifacts  # Expose as public.
-        ...
+
+        self.artifacts = self.manager._artifacts
+        self._meta = defaultdict(dict)  # to be exported as JSON at the end
+        self._meta['metadata']['descriptors'] = defaultdict(lambda:
+                                                            defaultdict(dict))
+        self._stream_names = {}  # maps stream_names to each descriptor uids
+        self._files = {}  # map descriptor uid to file handle of tiff file
+        self._filenames = {}  # map descriptor uid to file names of tiff files
+        self._file_prefix = file_prefix
+        self._templated_file_prefix = ''
+        self._kwargs = dict(cls=cls, **kwargs)
+
+    def start(self, doc):
+        '''Add `start` document information to the metadata dictionary.
+        This method adds the start document information to the metadata
+        dictionary. In addition it checks that only one `start` document is
+        seen.
+        Parameters:
+        -----------
+        doc : dict
+            RunStart document
+        '''
+
+        # raise an error if this is the second `start` document seen.
+        if 'start' in self._meta['metadata']:
+            raise RuntimeError(
+                "The serializer in suitcase.json expects documents from one "
+                "run only. Two `start` documents where sent to it")
+
+        # add the start doc to self._meta and format self._file_prefix
+        self._meta['metadata']['start'] = doc
+        self._templated_file_prefix = self._file_prefix.format(**doc)
+
+    def stop(self, doc):
+        '''Add `stop` document information to the metadata dictionary.
+        This method adds the stop document information to the metadata
+        dictionary. In addition it also creates the metadata '.json' file and
+        exports the metadata dictionary to it.
+        Parameters:
+        -----------
+        doc : dict
+            RunStop document
+        '''
+        # add the stop doc to self._meta.
+        self._meta['metadata']['stop'] = doc
+
+        # open a json file for the metadata and add self._meta to it.
+        f = self.manager.open('run_metadata',
+                              f'{self._templated_file_prefix}meta.json', 'xt')
```

Comment:
I would argue yes, I think it will be used predominantly in conjunction with other suitcases and therefore having `meta` as part of the filename would be helpful.